nixos/modules/services/misc/home-assistant.nix

```diff
@@ -245,22 +245,71 @@ in {
         rm -f "${cfg.configDir}/ui-lovelace.yaml"
         ln -s ${lovelaceConfigFile} "${cfg.configDir}/ui-lovelace.yaml"
       '');
-      serviceConfig = {
+      serviceConfig = let
+        # List of capabilities to equip home-assistant with, depending on configured components
+        # Converted to a string, so an empty list would not lead to no capability bounding set at all
+        capabilities = toString (unique (optionals (useComponent "bluetooth_tracker" || useComponent "bluetooth_le_tracker") [
+          # Required for interaction with hci devices and bluetooth sockets
+          # https://www.home-assistant.io/integrations/bluetooth_le_tracker/#rootless-setup-on-core-installs
+          "CAP_NET_ADMIN"
+          "CAP_NET_RAW"
+        ] ++ lib.optionals (useComponent "emulated_hue") [
+          # Alexa looks for the service on port 80
+          # https://www.home-assistant.io/integrations/emulated_hue
+          "CAP_NET_BIND_SERVICE"
+        ] ++ lib.optionals (useComponent "nmap_tracker") [
+          # https://www.home-assistant.io/integrations/nmap_tracker#linux-capabilities
+          "CAP_NET_ADMIN"
+          "CAP_NET_BIND_SERVICE"
+          "CAP_NET_RAW"
+        ]));
```

Comment:
FWIW: I came up with this list by grepping the home-assistant website.

We've covered the bluetooth and nmap trackers as well as emulated_hue.

We don't have support for `ps4`, `eddystone_temperature` and `kulersky`

I'm not quite sure how we can handle the capabilities on `hcitool`, is anyone using bluetooth trackers?

```
❯ rg "\\bcap_"
source/_integrations/bluetooth_le_tracker.markdown
80:sudo setcap 'cap_net_raw,cap_net_admin+eip' $(readlink -f $(which python3))
81:sudo setcap 'cap_net_raw+ep' $(readlink -f $(which hcitool))

source/_integrations/dwd_weather_warnings.markdown
36:A list of valid warncell ids and names can be found at https://www.dwd.de/DE/leistungen/opendata/help/warnungen/cap_warncellids_csv.html.

source/_integrations/emulated_hue.markdown
170:sudo setcap 'cap_net_bind_service=+ep' /srv/homeassistant/homeassistant_venv/bin/python3

source/_integrations/eddystone_temperature.markdown
28:sudo setcap 'cap_net_raw,cap_net_admin+eip' $(readlink -f $(which python3))

source/_integrations/nmap_tracker.markdown
95:sudo setcap cap_net_raw,cap_net_admin,cap_net_bind_service+eip /usr/bin/nmap

source/_integrations/kulersky.markdown
38:sudo setcap 'cap_net_raw,cap_net_admin+eip' `readlink -f \`which python3\``
39:sudo setcap 'cap_net_raw+ep' `readlink -f \`which hcitool\``

source/_integrations/ps4.markdown
67:sudo setcap 'cap_net_bind_service=+ep' <python>
75:sudo setcap 'cap_net_bind_service=+ep' /usr/bin/python3.5
```

Reply:
I gave the `nmap_tracker` a shot and it looks to be working like this.